charts/opentelemetry-collector/templates/daemonset.yaml

```
@@ -10,6 +10,10 @@ spec:
     matchLabels:
       {{- include "opentelemetry-collector.selectorLabels" . | nindent 6 }}
       component: agent-collector
+  updateStrategy:
+    rollingUpdate:
+      {{- include "opentelemetry-collector.rollingUpdate" . | nindent 6 }}
```

I wouldn't recommend overriding k8s's defaults

```suggestion
    {{- if or .Values.agentCollector.rollingUpdate.maxSurge .Values.agentCollector.rollingUpdate.maxUnavailable }}
    rollingUpdate:
      {{- if .Values.agentCollector.rollingUpdate.maxSurge }}
      maxSurge: {{ .Values.agentCollector.rollingUpdate.maxSurge }}
      {{- end }}
      {{- if .Values.agentCollector.rollingUpdate.maxUnavailable }}
      maxSurge: {{ .Values.agentCollector.rollingUpdate.maxUnavailable }}
      {{- end }}
    {{- end }}
```

---

Adjusted, but, to keep default values required, i did a little different